components/IscnUploadForm.vue

```diff
@@ -472,7 +577,13 @@ export default class UploadForm extends Vue {
   }
 
   handleDeleteFile(index: number) {
-    this.fileRecords.splice(index, 1)
+    const deletedFile = this.fileRecords[index];
+    this.fileRecords.splice(index, 1);
+
+    const indexToDelete = this.epubMetadataList.findIndex(item => item.epubFileName === deletedFile.fileName);
```

Comment:
should we assign unique random id to these items instead of using `fileName`? `fileName` might not be unique